opt/wireguard/interface.sh

```diff
@@ -0,0 +1,135 @@
+    # If address is deleted
+    if [ "$ACTION" = DELETE ]; then
+        OP=delete
+    else
+        OP=add
+    fi
+    # Parse all IP address on interface
+    for ip in $(ip a show dev $INTERFACE | grep inet | awk '{print $2}'); do
+        # If adding IP address to the interface and IP address is already setup on interface
+        if [ $OP == "add" ] && [ $ip == "$1" ]; then
+            # Do not process the rest of the function
+            return
+        fi
+    done
+    # Execute operation on link
+    sudo /opt/vyatta/sbin/vyatta-address $OP $INTERFACE $1
+}
+function cfg_description {
+    # If description has value
+    if node_exists description; then
+        # Set link alias
+        ip link set dev $INTERFACE alias "$(node_value description)"
+    else
+        # Remove link alias
+        sudo sh -c "echo > /sys/class/net/$INTERFACE/ifalias"
+    fi
+}
+function cfg_fwmark {
+    # If fwmark has value
+    if node_exists fwmark; then
+        # Mark packets leaving this interface
+        sudo wg set $INTERFACE fwmark $(node_value fwmark)
+    else
+        # Do not mark packets leaving this interface
+        sudo wg set $INTERFACE fwmark 0
+    fi
+}
+function cfg_listen-port {
+    # If listen-port has value
+    if node_exists listen-port; then
+        # Set listen-port
+        sudo wg set $INTERFACE listen-port $(node_value listen-port)
+    else
+        # Set listen-port to random port
+        sudo wg set $INTERFACE listen-port 0
+    fi
+}
+function cfg_mtu {
+    # If mtu has value
+    if node_exists mtu; then
+        # Set link MTU
+        ip link set $INTERFACE mtu $(node_value mtu)
+    fi
+}
+function cfg_private-key {
+    # If private-key has value
+    if node_exists private-key; then
+        # Create variable for private-key value
+        PRIVATE_KEY=$(node_value private-key)
+        # If private-key is a file
+        if [ -f "$PRIVATE_KEY" ]; then
+            # Set private-key to file
+            sudo wg set $INTERFACE private-key $PRIVATE_KEY
+        else
+            # Set private-key to value
+            echo $PRIVATE_KEY | sudo wg set $INTERFACE private-key /proc/self/fd/0
+        fi
+    else
+        # Remove private-key
+        sudo wg set $INTERFACE private-key /dev/null
+    fi
+}
+function cfg_route-allowed-ips {
+    # Update routing table
+    /opt/wireguard/update_routes.sh $INTERFACE
+}
+
+## Interface option configuration
+# If more than two parameters are passed to this script
+if [ $# -gt 2 ]; then
+    # If function exists in script, then run the function
+    type cfg_$3 2> /dev/null | grep -q 'function' && eval "cfg_$3 ${4:-}"
+    # Do not process the rest of the script
+    exit
+fi
+
+## Main interface configuration
+# If link doesn't exist
+if ! ip link show dev $INTERFACE &> /dev/null; then
+    # Create link
+    sudo ip link add dev $INTERFACE type wireguard
+fi
+
+# Run all configured 'down' commands
+eval "$(node_value down-command)" > /dev/null || exit 1
```

Comment:
I believe this should be added as an else condition to the previous if then statement. We should not have to run the down commands on a new interface.

Reply:
@FossoresLP What is your take on this?

Reply:
I agree. When the interface is only then created we should not have to run down commands even when the interface is deleted afterwards.
Do you want to create a PR or should I just add that myself?
A new release is more or less necessary anyway since I messed up the package release in the last one.